### ISSUE TITLE
ddd token no longer parses

### ISSUE THREAD

**Issue by jlowin** [crsmithdev/arrow#273]
Consider this code (based on getting the last modified date of an S3 key):

``` python
import arrow
dt = 'Sat, 10 Oct 2015 08:01:18 GMT'
dt_fmt = 'ddd, D MMM YYYY HH:mm:ss'
arrow.get(dt, dt_fmt)
```

After upgrading an old version of arrow, it started raising errors:

```
Traceback (most recent call last):
  File "/Users/jlowin/Desktop/arrow_test.py", line 4, in <module>
    arrow.get(dt, dt_fmt)
  File "/Users/jlowin/git/arrow/arrow/api.py", line 23, in get
    return _factory.get(*args, **kwargs)
  File "/Users/jlowin/git/arrow/arrow/factory.py", line 198, in get
    dt = parser.DateTimeParser(locale).parse(args[0], args[1])
  File "/Users/jlowin/git/arrow/arrow/parser.py", line 138, in parse
    raise ParserError('Failed to match \'{0}\' when parsing \'{1}\''.format(fmt_pattern, string))
arrow.parser.ParserError: Failed to match 'ddd, (?P<D>\d{1,2}) (?P<MMM>(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)) (?P<YYYY>\d{4}) (?P<HH>\d{2}):(?P<mm>\d{2}):(?P<ss>\d{2})' when parsing 'Sat, 10 Oct 2015 08:01:18 GMT'
```

According to `git bisect`, it was broken by commit a33a1eff50087e725bdfe82a51caf79275c99c17 from #249.

**Comment by pombredanne** [comment]
@jlowin looking at the code, I think that parsing ddd and dddd was never actually implemented at all and the day string was systematically ignored. There is not even a single test for it.
I was somehow giving results because the 10 Oct date was being used instead.

**Comment by pombredanne** [comment]
@crsmithdev is this by design to avoid conflict between the declared day and a possible date in digit?
or an oversight?

**Comment by pombredanne** [comment]
@beenje it looks like it is your commit that broke this alright

**Comment by jlowin** [comment]
Interesting! I think you're right -- it's only used for formatting, not parsing. At a minimum I think the docs should be changed to reflect that, since it includes it in the list of tokens for both formatting and parsing. I've since realized that the benefit of @beenje's regex parser is that I can safely drop the 'ddd' from my parse string. However, I do think tokens should be uniform across the two functions (format/parse) so that dates can be trivially converted with a single format string.

**Comment by andrewelkins** [comment]
They should be the same. I think it would be confusing to have different tokens for parsing vs formatting. Anyone have a moment to fix it via a PR. I don't think I'll get to it too quickly.

**Comment by pombredanne** [comment]
@andrewelkins @jlowin the addition of parsing ddd and dddd should be reasonably easy with some caveats... what if there is a inconsistency between a day in letters and a day in digit in the same date? which one should win? should there be an exception thrown?

**Comment by jlowin** [comment]
Personally, I think that digits should always beat characters, since digits are always more precise than names (the one "10th" day of a month vs the four "Wednesdays" of a month") 

However, how do you parse "Sat, October 2015"? Do we take the first Saturday of October, analogous to how "October 2015" yields the first day of October? It might be easier if `ddd` and `dddd` are recognized as valid tokens (i.e. they don't raise an error if they are present, which is the current behavior) but they don't actually affect the date parsing.

If they do affect the date parsing, then I would raise a `ValueError` if they are both present but don't agree.

**Comment by andrewelkins** [comment]
@pombredanne  I think if it's a mismatch then it should throw an error. I don't like the hiding of how it's parsing.

@jlowin I agree with the ValueError being thrown.

**Comment by swistakm** [comment]
Had the same issue today after upgrade so I'm sure this is kind of regression despite it was not tested. Note that in most cases `ddd` and `dddd` tokens do not need to be parsed but at least accepted. Note that @jlowin posted as his example is an [RFC 822](http://www.w3.org/Protocols/rfc822/#z28) date time formats that is quite popular. Most of all it is used in:
* email headers
* RSS/MRSS `<pubDate/>` elem

So it really makes sense to be able to handle those tokens somehow. Previously it worked as @jlowin proposed here: recognised but not parsed. It would be useful to revert such change. Such exceptional behaviour that can be easily documented seems like a sensible compromise for me. As far as I remember there is already a footnote in tokens table saying that `ddd`, `dddd` are used for format only.

Anyway I suppose @jlowin you found workaround for your issue. If not here is one that works for me:
```python
arrow.get('Fri, 22 May 2015 08:15:00 +0200', [r"\w+, DD MMM YYYY HH:mm:ss Z"], locale='en')
```
Because format strings are simply preprocessed regexes. By the way: you example used GMT as a timezone specifier and there is no proper token for such, but I think there could be separate issue (feature request) use for such.

**Comment by muffl0n** [comment]
I just came across the same problem. As this is already fixed in PR #321 it would be sweet if there was a new release that includes this fix. 
Thank you very much!

**Comment by andrewelkins** [comment]
Fixed by #321
